### PR TITLE
Add Dockerfile healthcheck

### PR DESCRIPTION
**Pull request by katosdev**
## What type of PR is this?

Enhancement

## What this PR does / why we need it:

Enhancement to add Dockerfile health check

## Special notes for your reviewer:

Need to confirm the BUSYBOX Version, not sure if this is the best way to handle this, so open to suggestions!

## Summary by CodeRabbit

- **New Features**
  - Added health check functionality using `wget` to monitor application status in Docker images.
  
- **Improvements**
  - Updated Docker images to include the `wget` tool for enhanced functionality and monitoring.
  - Changed the base image to `gcr.io/distroless/java:latest` for improved security and performance.

### CODE REVIEW

**Comment by katosdev** [comment]
@coderabbitai review

File: Dockerfile.rootless [hunk 1]
```diff
@@ -30,6 +31,14 @@ RUN CGO_ENABLED=0 GOOS=linux go build \
...
+# Downloading Wget
+ADD https://busybox.net/downloads/binaries/$BUSYBOX_VERSION/busybox_WGET /wget
+RUN chmod a+x /wget
+
+FROM gcr.io/distroless/java
+
+COPY --from=builder /wget /usr/bin/wget
+
 # Production Stage
 FROM gcr.io/distroless/static
```
[the remark this answers is not in the document]
Comment (katosdev) [reply Dockerfile.rootless]: Very good point actually, we should probably do this. Thoughts @tankerkiller125 ? Can you see this causing any issues? 😄

File: Dockerfile.rootless [hunk 2]
```diff
@@ -30,6 +31,14 @@ RUN CGO_ENABLED=0 GOOS=linux go build \
     # create a directory so that we can copy it in the next stage
     mkdir /data
 
+# Downloading Wget
+ADD https://busybox.net/downloads/binaries/$BUSYBOX_VERSION/busybox_WGET /wget
+RUN chmod a+x /wget
+
+FROM gcr.io/distroless/java
```
[the remark this answers is not in the document]
Comment (katosdev) [reply Dockerfile.rootless]: Thanks @coderabbitai - This has now been amended.

File: Dockerfile.rootless (hunk 1 above)
[the remark this answers is not in the document]
Comment (katosdev) [reply Dockerfile.rootless]: @coderabbitai change committed.

File: Dockerfile.rootless (hunk 2 above)
[the remark this answers is not in the document]
Comment (katosdev) [reply Dockerfile.rootless]: @coderabbitai resolve

File: Dockerfile.rootless (hunk 1 above)
[the remark this answers is not in the document]
Comment (katosdev) [reply Dockerfile.rootless]: @coderabbitai resolve

**Comment by katosdev** [comment]
@coderabbitai review

**Comment by katosdev** [comment]
@coderabbitai full review

**Comment by tankerkiller125** [comment]
Don't forget you also need to update the `Dockerfile` @katosdev, right now only the rootless one would have a healthcheck.

**Comment by katosdev** [comment]
> Don't forget you also need to update the `Dockerfile` @katosdev, right now only the rootless one would have a healthcheck.

Indeed, was making sure that our bot overlords weren’t going to rage at me, and then I’ll spin this into the dockerfile also. 🙂

**Comment by katosdev** [comment]
@coderabbitai full review

**Pull request merged**